src/rename.c

```diff
@@ -446,6 +446,43 @@ void rename_detection(struct manifest *manifest)
 	g_list_free(deleted_files);
 }
 
+void final_link(GList *files)
+{
+	GList *list1, *list2;
+	struct file *file1, *file2;
+
+	list1 = new_list_renamed_files(files);
+	list1 = list2 = g_list_sort(list1, file_sort_version);
+
+	for (; list1; list1 = g_list_next(list1)) {
+		file1 = list1->data;
+		/* check the rest of the list */
+		list2 = g_list_next(list1);
+		for (; list2; list2 = g_list_next(list2)) {
+			file2 = list2->data;
+			/* must have same hash to link */
+			if (!hash_compare(file2->hash, file1->hash)) {
+				continue;
+			}
+
+			/* must be the same version to link */
+			if (file1->last_change != file2->last_change) {
```

Comment:
Since you are sorting by version, it would probably be better to check this condition first (before the hash_compare).

Reply:
yes, good point.